cql3/restrictions/single_column_primary_key_restrictions.hh

```diff
@@ -187,6 +187,8 @@ public:
                             join(", ", get_column_defs())));
         }
         do_merge_with(::static_pointer_cast<single_column_restriction>(restriction));
+        this->wip_equivalent = ::make_shared<wip::expression>(
+                wip::conjunction{std::vector{this->wip_equivalent, restriction->wip_equivalent}});
```

Comment:
Again list->tree. I suggest a helper make_conjunction(expression&&, expression&&) that peeks into the input types and flattens them if one or both are already a conjunction. This is not so much as a premature optimization[1] as to preserve the existing structure and not to disturb any assumptions the code makes.

[1] ok, I admit that I can't see a list being converted to a tree

Reply:
Done.